agent/grpc-external/services/resource/write.go

```diff
@@ -37,6 +37,10 @@ func (s *Server) Write(ctx context.Context, req *pbresource.WriteRequest) (*pbre
 		)
 	}
 
+	if err = reg.Validate(req.Resource); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "failed validation: %v", err)
```

Comment:
I wonder if it might be better to remove the `failed validation: ` prefix, as our own structural validation (e.g. checking `resource.id` is non-nil) doesn't have it?